### ISSUE TITLE
Response patching for POST does not work

### ISSUE THREAD

**Issue by skipdavis1117** [mswjs/msw#77]
Using ctx.fetch inside the response resolver for a POST results in a GET being sent (with no body) instead of a POST (with a body).

Notes:
- The req before sending the original request showed the correct body and bodyUsed:true, but looking at the req in mockServiceWorker showed the body gone.  
I suspect the problem is in context/fetch.ts, but initial scan did not show an obvious issue to me (JavaScript beginner)

I wrote an integration test to PR into the repo that replicated the issue, but I did not have permission to push up a branch.

## Tests:

```ts
// response-mocking.mocks.ts:
 rest.post(
  'https://jsonplaceholder.typicode.com/posts',
  async (req, res, ctx) => {
    const originalResponse = await ctx.fetch(req)
    return res(
      ctx.json({
        ...originalResponse,
        mocked: true,
      }),
    )
  },
),
```
```ts
// response-patching.test.ts:
describe('given a post request to be patched', () => {
  it('should be able to properly request and patch a post', async () => {
    const REQUEST_URL = 'https://jsonplaceholder.typicode.com/posts'

    api.page.evaluate(
      (url) =>
        fetch(url, {
          method: 'POST',
          headers: {
            'Content-Type': 'application/json',
          },
          body: JSON.stringify({
            title: 'foo',
            body: 'bar',
            userId: 1,
          }),
        }),
      REQUEST_URL,
    )
    const res = await api.page.waitForResponse((res) => {
      return (
        // Await for the response from MSW, so that original response
        // from the same URL would not interfere.
        res.url() === REQUEST_URL && res.headers()['x-powered-by'] === 'msw'
      )
    })
    const body = await res.json()

    expect(res.status()).toBe(200)
    expect(body).toEqual({
      id: 101,
      title: 'foo',
      body: 'bar',
      userId: 1,
      mocked: true,
    })
  })
})

**Comment by kettanaito** [comment]
Thanks for reporting this, @skipdavis1117! The intent is that `ctx.fetch(req)` performs the very same request that it got as an argument. If there is a change of method/body/any other request property taking place that's an unexpected behavior indeed.

Your test looks good. Please, if you could fork this repository and create a new branch in your fork, then you could push the new test to your branch and issue a pull request to this repo directly. Then I'd be able to look into this using your test suite.

**Comment by skipdavis1117** [comment]
Will do.  Thanks!

**Comment by kettanaito** [comment]
Released in [`0.11.4`](https://github.com/open-draft/msw/releases/tag/v0.11.4).

Thanks for help, @skipdavis1117! Could you please update and verify that response patching for POST requests works for you now?
